guides/cookbook.md

```diff
@@ -169,6 +170,59 @@ int main(int argc, char **argv) {
 ```
 </div>
 
+<div class="tabpane C-sharp" markdown="1">
+Implementations of the IIonReader and IIonWriter interfaces are responsible for reading and writing
+Ion data in both text and binary forms.
+
+The following example shows how text Ion data can be read from a string:
+```c#
+using IonDotnet;
+using IonDotnet.Builders;
+using System;
+
+class ReadIonData
+{
+    static void Main(string[] args)
+    {
+        IIonReader reader = IonReaderBuilder.Build("{hello: \"world\"}");
+        reader.MoveNext();                           // position the reader at the first value, a struct
+        reader.StepIn();                             // step into the struct
+        reader.MoveNext();                           // position the reader at the first value in the struct
+        string fieldName = reader.CurrentFieldName;  // retrieve the current value's field name
+        string value = reader.StringValue();         // retrieve the current value's string value
+        reader.StepOut();                            // step out of the struct
+        Console.WriteLine(fieldName + " " + value);  // prints:  hello world
+    }
+}
+```
+
+In the above example, the text Ion `{hello: "world"}` was probably typed by a human using a text
+editor. The following example illustrates how it could have been generated using an IIonWriter:
+
+```c#
+using IonDotnet;
+using IonDotnet.Builders;
+using System;
+using System.IO;
+
+class WriteIonText
+{
+    static void Main(string[] args)
+    {
+        TextWriter tw = new StringWriter();
+        IIonWriter writer = IonTextWriterBuilder.Build(tw);
+        writer.StepIn(IonType.Struct);     // step into a struct
+        writer.SetFieldName("hello");      // set the field name for the next value to be written
+        writer.WriteString("world");       // write the next value
+        writer.StepOut();                  // step out of the struct
+        writer.Finish();
+        Console.WriteLine(tw.ToString());  // prints:  {hello:"world"}
+    }
+}
+```
+
+If the desired output is pretty text, pass an instance of `IonTextOptions` with `PrettyPrint` set to `true` to `IonTextWriterBuilder.Build()`.    If Ion binary encoding is desired, use `IonBinaryWriterBuilder` (instead of `IonTextWriterBuilder`).
```

Comment:
Is the first sentence necessary, given that we have explanation and an example of pretty printing below?

Reply:
Agreed, and I like the idea of keeping each example focused on a relatively narrow set of concepts.  I've removed information about pretty-printing from the corresponding JS example as well.